### ISSUE TITLE
Avatar creator and inspectors are broken in a certain world

### ISSUE THREAD

**Issue by toritoriVR** [Yellow-Dog-Man/Resonite-Issues#1329]
### Describe the bug?

When using Avatar Creator in a specific world, it is broken and does not work properly as shown in the attached image. 
![bagu01](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/94087561/530de97d-e018-4c67-8fdb-0683c4a07c3b)
![bug2](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/94087561/571f138e-5eec-4d6e-9c3f-e6b34794373a)

### To Reproduce

When you start Avatar Creator in this world, everyone will be recreated 100%.
resrec:///G-toritori-group/R-ca2e13c0-edc1-4e98-a65a-90f6a0e42b02
https://go.resonite.com/record/G-toritori-group/R-ca2e13c0-edc1-4e98-a65a-90f6a0e42b02

### Expected behavior

You cannot set up an avatar in this world because the button is missing from the creator. The cause is unknown. It seems that the world migrated from Neos was working normally at the beginning of the migration.

### Resonite Version Number

2024.2.5.848

### What Platforms does this occur on?

Windows

### Log Files

[TYPE81 - 2024.2.5.848 - 2024-02-11 22_20_35.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/14232499/TYPE81.-.2024.2.5.848.-.2024-02-11.22_20_35.log)

### Additional Context

The symptoms started about a month ago, and I don't know if it's due to the world change or client update.

### Reporters

username:toritori  DiscordID:とり#0890

**Comment by ohzee00** [comment]
This is perhaps the most broken I've seen a world and I have no idea what's causing it. Inspectors in that world do not fully work either, making this really hard to pin point. 

(I deleted everything in the world to see if something was causing it but the issue still persisted)
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/33611545/2a6a2ad4-5942-4540-90d3-b1c3ac6b3a9b)

**Comment by ohzee00** [comment]
This issue still persisted on saving the world in my own inventory and opening it. However the world is still salvageable if you put all the world's slots into one slot and save it so you can transfer it to another world. 
I had difficulty saving world via a single slot due to some avatars (I'm assuming) having avatar protection, but the inspectors/avatar creator issue did not follow into another world when transferring it like that.

It's so strange, at first I thought it was a weird material issue being override but the inspector straight up doesn't build itself fully in that world. You don't even get a hitch when opening up root or anything.

Though I would suggest renaming this issue to "Avatar creator and inspectors are broken in a certain world" or similar.

**Comment by toritoriVR** [comment]
Thank you ohzee00 for the verification. I have changed the title according to your advice.

**Comment by Nytra** [comment]
Looking at the log file there is the error `System.Exception: An incompatible component is stored under the key BasicUI_Check`

This seems to be preventing any UIX which uses a checkbox from being created, which also prevents the inspectors from working correctly.

So somehow the key for that component has been broken or corrputed in this world.

I actually do remember there was another issue with a similar problem causing UI to not work correctly in a world: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1073

Full exception:

```
22:22:02.838 ( 90 FPS)	Exception running OnAttach behavior for Component: Element: ID41CBE00, Type: FrooxEngine.AvatarCreator, World: アバターインポートを比較的丁寧に
説明するワールド, IsRemoved: False, IsDestroyed: False, IsDisposed: False, Enabled: True
Element: ID41CB300, Type: FrooxEngine.Slot, World: アバターインポートを比較的丁寧に
説明するワールド, IsRemoved: False, Slot name: Avatar Creator, T: [25.1937; 1.617754; -24.31787], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID2400, Type: FrooxEngine.Slot, World: アバターインポートを比較的丁寧に
説明するワールド, IsRemoved: False, Slot name: Root, T: [0; 0; 0], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID0, Type: FrooxEngine.World, World: アバターインポートを比較的丁寧に
説明するワールド, IsRemoved: False

System.Exception: An incompatible component is stored under the key BasicUI_Check
Exiting: Element: ID37400, Type: FrooxEngine.OverlayFresnelMaterial, World: アバターインポートを比較的丁寧に
説明するワールド, IsRemoved: False, IsDestroyed: False, IsDisposed: False, Enabled: True
Element: ID2400, Type: FrooxEngine.Slot, World: アバターインポートを比較的丁寧に
説明するワールド, IsRemoved: False, Slot name: Root, T: [0; 0; 0], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID0, Type: FrooxEngine.World, World: アバターインポートを比較的丁寧に
説明するワールド, IsRemoved: False

  at FrooxEngine.SharedComponents.GetSharedComponentOrCreate[T] (FrooxEngine.World world, System.String sharedKey, System.Action`1[T] onCreate, System.Int32 version, System.Boolean replaceExisting, System.Boolean updateExisting, System.Func`1[TResult] getRoot) [0x00067] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.UIX.UIBuilder.get_CheckSprite () [0x00014] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.UIX.UIBuilder.Checkbox (System.Boolean state) [0x000b4] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.UIX.UIBuilder.Checkbox (Elements.Core.LocaleString& label, FrooxEngine.UIX.Text& labelText, System.Boolean state, System.Boolean labelFirst, System.Single padding) [0x00093] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.UIX.UIBuilder.Checkbox (Elements.Core.LocaleString& label, System.Boolean state, System.Boolean labelFirst, System.Single padding) [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.AvatarCreator.OnAttach () [0x009a7] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.ComponentBase`1[C].RunOnAttach () [0x0001a] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 

  at System.Environment.get_StackTrace () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Elements.Core.UniLog.Error (System.String message, System.Boolean stackTrace) [0x00000] in <af14e8ef2eea4b7b988c02dcc112935e>:0 
  at FrooxEngine.DebugManager.Error (System.Object obj, System.Boolean stackTrace) [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.ComponentBase`1[C].RunOnAttach () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.ContainerWorker`1[C].AttachComponentInternal (C component, System.Boolean runOnAttachBehavior, System.Action`1[T] beforeAttach) [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.ContainerWorker`1[C].AttachComponent[T] (System.Boolean runOnAttachBehavior, System.Action`1[T] beforeAttach) [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.AvatarCreatorSpawner+<>c__DisplayClass0_0.<Pressed>b__0 () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.World.RunSynchronousActions () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.World.NextRefreshStage () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.World.Refresh () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.WorldManager.UpdateStep () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.WorldManager.RunUpdateLoop () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.Engine.UpdateStep () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at FrooxEngine.Engine.RunUpdateLoop () [0x00000] in <a5645f6bd5bf42c5a2c17d81c56a6337>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.UpdateFrooxEngine () [0x00000] in <113dc9c88b744d84a5c748dff565acf6>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.Update () [0x00000] in <113dc9c88b744d84a5c748dff565acf6>:0 
```

**Comment by Frooxius** [comment]
It looks like some shared components got replaced by a wrong variant.

We can fix this without additional info.

**Comment by Frooxius** [comment]
This should be fixed in 2024.2.12.1430! If this still persists, let me know and we'll reopen!
